src/Files.App/UserControls/Widgets/RecentFilesWidget.xaml.cs

```diff
@@ -41,19 +41,23 @@ private async void RecentFilesListView_DragItemsStarting(object sender, DragItem
 				var storageItems = new List<IStorageItem>();
 				var fileContents = new List<string>();
 
-
 				foreach (var item in items)
 				{
 					try
 					{
-						// Attempt to get the file from its path
+						// Attempt to get the file from its path and read the file text content
 						var file = await StorageFile.GetFileFromPathAsync(item.RecentPath);
 						if (file != null)
+						{
 							storageItems.Add(file);
 
-						// Read the file text content
-						var fileContent = await FileIO.ReadTextAsync(file);
-						fileContents.Add(fileContent);
+							//Attempt to read the file text content
+							var fileContent = await FileIO.ReadTextAsync(file);
```

Comment:
Sorry to keep reviewing, but wouldn't this cause a memory squeeze in the case of a large text file? It would be better to set an upper limit on the file size to be read.